corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -50,22 +50,29 @@ hqDefine("data_dictionary/js/data_dictionary", [
 
                     var propObj = propertyListItem(prop.name, prop.label, false, group.name, self.name, prop.data_type,
                         prop.description, prop.allowed_values, prop.fhir_resource_prop_path, prop.deprecated,
-                        prop.removeFHIRResourcePropertyPath, isGeoCaseProp, prop.is_safe_to_delete, prop.id);
-                    propObj.description.subscribe(changeSaveButton);
-                    propObj.label.subscribe(changeSaveButton);
-                    propObj.fhirResourcePropPath.subscribe(changeSaveButton);
-                    propObj.dataType.subscribe(changeSaveButton);
-                    propObj.deprecated.subscribe(changeSaveButton);
-                    propObj.deleted.subscribe(changeSaveButton);
-                    propObj.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
+                        prop.removeFHIRResourcePropertyPath, isGeoCaseProp, prop.is_safe_to_delete, prop.id, prop.index, groupObj.name());
+                    subscribePropObservable(propObj, propObj.description, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.label, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.fhirResourcePropPath, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.dataType, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.deprecated, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.deleted, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.removeFHIRResourcePropertyPath, changeSaveButton);
                     propObj.allowedValues.on('change', changeSaveButton);
+                    propObj.allowedValues.on('change', propObj.allowedValuesChanged);
                     groupObj.properties.push(propObj);
                 }
                 groupObj.properties.subscribe(changeSaveButton);
                 self.groups.push(groupObj);
             }
         };
 
+        function subscribePropObservable(propObj, prop, changeSaveButton) {
+            prop.subscribe(changeSaveButton);
+            const valChangedFunc = (newVal) => propObj.valChanged(newVal, prop._oldValue);
```

Comment:
Hey @zandre-eng 
Is `_oldValue` maintained by knockoutJS? I don't see it getting used anywhere in the changes you have made.

Reply:
Strangely, this worked perfectly yesterday and I was under the impression that this was a KnockoutJS element after doing some research. But coming back to the code again today, this isn't working at all. I'm going to refactor this code so that this logic is a bit more robust and consistent.

Reply:
The logic has been refactored ([da37844](https://github.com/dimagi/commcare-hq/pull/34781/commits/da37844851659e8e50b44061a7cee9ecff04493f)). 

Now, we call a function which keeps track of the old value. The function also subscribes to any new changes to an observable, and then compares the old and new values to see if `hasChanges` should be made `true`.